app/src/main/java/de/xikolo/controllers/channels/ChannelDetailsFragment.kt

```diff
@@ -165,29 +164,35 @@ class ChannelDetailsFragment : NetworkStateFragment<ChannelViewModel>() {
 
     private fun enroll(courseId: String) {
         showBlockingProgress()
-        courseManager.createEnrollment(courseId, object : RequestJobCallback() {
-            public override fun onSuccess() {
-                if (view != null) {
-                    hideAnyProgress()
-                    val course = CourseDao.Unmanaged.find(courseId)
-                    if (course?.accessible == true) {
-                        enterCourse(courseId)
-                    }
-                }
-            }
 
-            public override fun onError(code: RequestJobCallback.ErrorCode) {
-                if (view != null) {
-                    hideAnyProgress()
-                    if (code === ErrorCode.NO_NETWORK) {
+        val enrollmentCreationNetworkState = NetworkStateLiveData()
```

Comment:
Why are those `LiveData` objects not managed by a `ViewModel`?

Reply:
Each `NetworkStateLiveData` can only be used once for such a request, because when you try to run a new request with an already used `NetworkStateLiveData`, once you call `observe(...)` on it, the `LiveData` object considers a notify with the previous state (which is often `SUCCESS`).

Reply:
Doesn't sound perfect but might be fine for now.